app/router/user_session.py

```diff
@@ -4,14 +4,29 @@
 from datetime import datetime
 from routes import user_session_db_url
 from helpers import db_request_token, is_response_valid, is_response_empty
+from router import student, session, user
+from request import build_request
 
 router = APIRouter(prefix="/user-session", tags=["User-Session"])
 
 
 @router.post("/")
-def user_session(user_session: UserSession):
+async def user_session(user_session: UserSession):
     query_params = user_session.dict()
     query_params["timestamp"] = datetime.now().isoformat()
+
+    if query_params["user_type"] == "student":
+        user_id_response = student.get_students(
+            build_request(query_params={"student_id": query_params["user_id"]})
+        )
+
+    query_params["user_id"] = user_id_response[0]["user"]["id"]
```

Comment:
check if user_id_response length > 0

Reply:
It will always be, no, because this call is made always after authentication only.